docs/src/pages/_App.js

```diff
@@ -0,0 +1,378 @@
+        setPendingFileContents(result);
+        setLoadFrom('file');
+      });
+      reader.readAsText(acceptedFiles[0]);
+    }
+  }, []);
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({ onDrop, maxFiles: 1});
+
+  useEffect(() => {
+    let unmounted = false;
+    async function processParams() {
+      if (url) {
+        setLoading(true);
+        try {
+          const response = await fetch(url);
+          if(unmounted) {
+            return;
+          }
+          if(response.ok) {
+            const responseText = await response.text();
+            if(unmounted) {
+              return;
+            }
+            if(edit) {
+              // User wants to edit the URL-based config.
+              try {
+                // Ideally, this is valid JSON and we can
+                // use JSON.stringify to add nice indentation.
+                const responseJson = JSON.parse(responseText);
+                setPendingConfig(JSON.stringify(responseJson, null, 2));
+              } catch(e) {
+                // However, this may be an invalid JSON object
+                // so we can just let the user edit the unformatted string.
+                setPendingConfig(responseText);
+              }
+              setError(null);
+            }
+            setLoading(false);
+          } else {
+            setError({
+              title: "Fetch response not OK",
+              message: response.statusText,
+            });
+            setLoading(false);
+            setPendingConfig('{}');
+          }
+        } catch(e) {
+          setError({
+            title: "Fetch error",
+            message: e.message,
+          });
+          setLoading(false);
+          setPendingConfig('{}');
+        }
+      } else if(demo && configs[demo]) {
+        setPendingConfig(JSON.stringify(configs[demo], null, 2));
+        setError(null);
+        setLoading(false);
+      } else {
+        setPendingConfig(baseJson);
+        setError(null);
+        setLoading(false);
+      }
+    }
+    processParams();
+    return () => {
+      unmounted = true;
+    };
+  }, [url, edit, demo]);
+
+  function validateConfig(nextConfig) {
+    const [failureReason, upgradeSuccess] = upgradeAndValidate(JSON.parse(nextConfig));
+    return [upgradeSuccess, failureReason];
+  }
+
+  function handleEditorGo() {
+    let nextUrl;
+    if(loadFrom === 'editor') {
+      let nextConfig = pendingConfig;
+      if(syntaxType === "JS") {
+        nextConfig = window[JSON_TRANSLATION_KEY];
+      }
+      nextUrl = 'data:,' + encodeURIComponent(nextConfig);
+
+      const [valid, failureReason] = validateConfig(nextConfig);
+      if(!valid) {
+        setError(failureReason);
+        return;
+      }
+    } else if(loadFrom === 'url') {
+      nextUrl = pendingUrl;
+    } else if(loadFrom === 'file') {
+      nextUrl = 'data:,' + encodeURIComponent(pendingFileContents);
+    }
+    window.location.href = baseUrl + nextUrl;
+  }
```

Comment:
Is there no better way to do this?  When I click the "load from editor" button, it does a full reload of the page - the page goes white for a second, then the homepage comes up, and finally the loaded view configuration.

I don't want to force a full refactor but it's kind of slow.

Reply:
To prevent breaking old `vitessce.io/?url=x` and `vitessce.io/?dataset=x` urls, we need to continue to support those query parameters here. I think the full reload is required because query parameters are designed to send parameters to the server (despite the lack of server-side usage here)

Related: limits on AWS S3 query parameter values cause 400 errors (when the `?url=x` value is too long):
> I also only just noticed that if I click EDIT on http://beta.vitessce.io/index.html?dataset=codeluppi-2018 then on the editor page I click LOAD FROM EDITOR (without having edited anything) I get a 400 error page.

One solution is to allow both query parameters and hash-based parameters:
- hash-based parameters like `vitessce.io/#url=x` will not need a full refresh and will not be sent to the server (so only subject to client-side limits)
- query parameter URLs like `vitessce.io/?url=x` will still work
   - maybe these should be deprecated and redirect to the hash-based equivalent?

Reply:
Hash param support added in https://github.com/vitessce/vitessce/pull/1088

Reply:
Thanks!  I think this makes sense.  I think the limit on query parameters is another motivating use-case for this, so I'm happy to see it done.